projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/buffer.cpp

```diff
@@ -219,7 +219,21 @@ flush(rocprofiler_buffer_id_t buffer_id, bool wait)
 
                     } catch(std::exception& e)
                     {
-                        ROCP_CI_LOG(ERROR) << "buffer callback threw an exception: " << e.what();
+                        ROCP_CI_LOG(ERROR) << fmt::format(
+                            "buffer callback threw an exception: {} [buffer_id={}, idx={}, "
+                            "offset={}, context_id={}, callback={}, callback_data={}, records={}, "
+                            "first_record_ptr={}, drop_count={}]",
+                            e.what(),
+                            buffer_id.handle,
+                            idx,
+                            offset,
+                            buff_v->context_id,
+                            reinterpret_cast<const void*>(buff_v->callback),
```

Comment:
```suggestion
                            static_cast<const void*>(buff_v->callback),
```
Casting to pointer to void pointer only requires static casting

Reply:
It is not allowed for function pointers:

> /root/rocm-systems/worktrees/pr1440/projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/buffer.cpp:231:29: error: static_cast from 'rocprofiler_buffer_tracing_cb_t' (aka 'void (*)(rocprofiler_context_id_t, rocprofiler_buffer_id_t, rocprofiler_record_header_t **, unsigned long, void *, unsigned long)') to 'const void *' is not allowed [clang-diagnostic-error]
                            static_cast<const void*>(buff_v->callback),

Reply:
Oh I wasn't paying enough attention and didn't realize it was function pointer